lib/pages/settings_dashboard/settings_profile/settings_profile.dart

```diff
@@ -0,0 +1,300 @@
+    debugLabel: 'displayNameFocusNode',
+  );
+
+  List<SettingsProfileEnum> get getListProfileMobile =>
+      getListProfileBasicInfo + getListProfileWorkIdentitiesInfo;
+
+  final List<SettingsProfileEnum> getListProfileBasicInfo = [
+    SettingsProfileEnum.displayName,
+  ];
+
+  final List<SettingsProfileEnum> getListProfileWorkIdentitiesInfo = [
+    SettingsProfileEnum.matrixId
+  ];
+
+  List<SheetAction<AvatarAction>> actions() => [
+        if (PlatformInfos.isMobile)
+          SheetAction(
+            key: AvatarAction.camera,
+            label: L10n.of(context)!.openCamera,
+            isDefaultAction: true,
+            icon: Icons.camera_alt_outlined,
+          ),
+        SheetAction(
+          key: AvatarAction.file,
+          label: L10n.of(context)!.openGallery,
+          icon: Icons.photo_outlined,
+        ),
+        if (profileNotifier.value.avatarUrl != null)
+          SheetAction(
+            key: AvatarAction.remove,
+            label: L10n.of(context)!.removeYourAvatar,
+            isDestructiveAction: true,
+            icon: Icons.delete_outlined,
+          ),
+      ];
+
+  TextEditingController? getController(
+    SettingsProfileEnum settingsProfileEnum,
+  ) {
+    switch (settingsProfileEnum) {
+      case SettingsProfileEnum.displayName:
+        return displayNameEditingController;
+      case SettingsProfileEnum.matrixId:
+        return matrixIdEditingController;
+      default:
+        return null;
+    }
+  }
+
+  FocusNode? getFocusNode(SettingsProfileEnum settingsProfileEnum) {
+    switch (settingsProfileEnum) {
+      case SettingsProfileEnum.displayName:
+        return displayNameFocusNode;
+      default:
+        return null;
+    }
+  }
+
+  void _handleRemoveAvatarAction() async {
+    final success = await showFutureLoadingDialog(
+      context: context,
+      future: () => matrix.client.setAvatar(null),
+    );
+    if (success.error == null) {
+      getCurrentProfile(client, isUpdated: true);
+    }
+    return;
+  }
+
+  Future<MatrixFile?> _handleGetAvatarInByte() async {
+    final result = await FilePicker.platform.pickFiles(
+      type: FileType.image,
+      withData: true,
+    );
+    final pickedFile = result?.files.firstOrNull;
+    if (pickedFile == null || pickedFile.bytes == null) return null;
+    return MatrixFile(
+      bytes: pickedFile.bytes!,
+      name: pickedFile.name,
+    );
+  }
+
+  Future<MatrixFile?> _handleGetAvatarInStream(AvatarAction action) async {
+    final result = await ImagePicker().pickImage(
+      source: action == AvatarAction.camera
+          ? ImageSource.camera
+          : ImageSource.gallery,
+      imageQuality: AppConfig.imageQuality,
+    );
+    if (result == null) return null;
+    return MatrixFile(
+      bytes: await result.readAsBytes(),
+      name: result.path,
+    );
+  }
+
+  void _handleGetAvatarAction(AvatarAction action) async {
+    MatrixFile file;
+    if (PlatformInfos.isMobile) {
```

Comment:
desktop can also use files i think

Reply:
no problem for desktop now. Change it later.